### DISCUSSION TITLE
Debugging Web Extension in PWA host with Extension Dependency always requires reload

### DISCUSSION THREAD

**Discussion by vritant24** [microsoft/vscode-discussions#854] [Extension Development QnA] [answered]
I have a web extension I'm trying to debug locally which depends on the [VSCode Wasm Wasi Core](https://marketplace.visualstudio.com/items?itemName=ms-vscode.wasm-wasi-core) extension. 

My lunch config has been set so to help debug it:
```json 
{
      "name": "Launch Web Extension ",
      "type": "pwa-extensionHost",
      "debugWebWorkerHost": true,
      "debugWebviews": true,
      "request": "launch",
      "sourceMaps": true,
      "args": [
        "--disable-extensions",
        "--extensionDevelopmentPath=${workspaceFolder}/vscode-extension",
        "--extensionDevelopmentPath=${userHome}/.vscode-insiders/extensions/ms-vscode.wasm-wasi-core-0.13.1",
        "--extensionDevelopmentKind=web"
      ],
      "outFiles": [
        "!**/node_modules/**",
        "${workspaceFolder}/vscode-extension/dist/**/*.js",
      ],
      "preLaunchTask": "Watch Full Web Extension"
    },
```

I'm using this way of debugging to force the wasm extension to run in the web environment as opposed to the desktop environment from following the discussion over [here](https://github.com/microsoft/vscode/issues/151815#issuecomment-1160982115) 

The issue ends up being that the wasm extension constantly asks for a reload no matter how many times the extension development vscode instance is reloaded
<img width="425" alt="image" src="https://github.com/microsoft/vscode-discussions/assets/13074644/c464fead-fd7c-4ee0-bf77-37e795977d2e">

Is there a way to work around this issue?

**Comment by isidorn** [comment]
The reload required button might have been a UI bug that we have fixed in the meantime.
Do you still see this with the latest VS Code insiders?
Does debugging actually work?

**Reply by vritant24** [reply]
It still runs into the same problem in the latest insiders.
Happy to provide more details if the info in the post is not enough to repro

**Reply by isidorn** [reply]
@connor4312 might be able to help

**Reply by connor4312** [reply]
I think this is more @sandy081's area

**Reply by sandy081** [reply]
Can you please click on that extension and let us know the reason for reload? Do you have that extension installed already?

**Reply by vritant24** [reply]
I do have the extension installed already.

This is what I see when I use the above launch config
![image](https://github.com/microsoft/vscode-discussions/assets/13074644/df82f6a0-cbe5-4b24-ba8f-6d0cceaaaa1f)

**Reply by SaadSaik0** [reply]
{
      "name": "Launch Web Extension ",
      "type": "pwa-extensionHost",
      "debugWebWorkerHost": true,
      "debugWebviews": true,
      "request": "launch",
      "sourceMaps": true,
      "args": [
        "--disable-extensions",
        "--extensionDevelopmentPath=${workspaceFolder}/vscode-extension",
        "--extensionDevelopmentPath=${userHome}/.vscode-insiders/extensions/ms-vscode.wasm-wasi-core-0.13.1",
        "--extensionDevelopmentKind=web"
      ],
      "outFiles": [
        "!**/node_modules/**",
        "${workspaceFolder}/vscode-extension/dist/**/*.js",
      ],
      "preLaunchTask": "Watch Full Web Extension"
    },

**Comment by sandy081** [comment, marked as answer]
It seems you have the following argument in your launch ` "--disable-extensions",` which is causing this confusion. Do you have this extension installed in general?

**Reply by vritant24** [reply]
I do have this extension installed in general.
I was under the assumption that disabling other extensions to remove noise would work alongside specifying the exact extensions to enable (to remove noise).

When I run the launch without `--disable-extension`, the `vscode-easm-wasi-core` seems to run, so that does unblock me. Thanks!

**Comment by sandy081** [comment]
You can also use profile if you do not want to have noise from other extensions while debugging - https://code.visualstudio.com/updates/v1_72#_extension-debugging-in-a-clean-environment

**Reply by waiswawicklif-commits** [reply]
Thank you

**Comment by SaadSaik0** [comment]
{
      "name": "Launch Web Extension ",
      "type": "pwa-extensionHost",
      "debugWebWorkerHost": true,
      "debugWebviews": true,
      "request": "launch",
      "sourceMaps": true,
      "args": [
        "--disable-extensions",
        "--extensionDevelopmentPath=${workspaceFolder}/vscode-extension",
        "--extensionDevelopmentPath=${userHome}/.vscode-insiders/extensions/ms-vscode.wasm-wasi-core-0.13.1",
        "--extensionDevelopmentKind=web"
      ],
      "outFiles": [
        "!**/node_modules/**",
        "${workspaceFolder}/vscode-extension/dist/**/*.js",
      ],
      "preLaunchTask": "Watch Full Web Extension"
    },
